src/oc/web/actions/activity.cljs

```diff
@@ -321,40 +341,50 @@
     (fn [{:keys [status success body]}]
       (dis/dispatch! [:activity-get/finish status (router/current-org-slug) (json->cljs body) nil]))))
 
-(defn entry-clear-local-cache [item-uuid edit-key]
+(declare entry-revert)
+
+(defn entry-clear-local-cache [item-uuid edit-key item]
   (remove-cached-item item-uuid)
+  ;; revert draft to old version
+  (timbre/debug "Reverting to " @initial-revision)
+  (when (not= "published" (:status item))
+    (entry-revert @initial-revision item))
   (dis/dispatch! [:entry-clear-local-cache edit-key]))
 
-(defn entry-save [edited-data & [section-editing edit-key]]
-  (let [fixed-edited-data (assoc edited-data :status (or (:status edited-data) "draft"))
-        fixed-edit-key (or edit-key :entry-editing)]
-    (if (:links fixed-edited-data)
-      (if (and (= (:board-slug fixed-edited-data) utils/default-section-slug)
-               section-editing)
-        (let [fixed-entry-data (dissoc fixed-edited-data :board-slug :board-name :invite-note)
-              final-board-data (assoc section-editing :entries [fixed-entry-data])]
-          (api/create-board final-board-data (:invite-note fixed-edited-data)
-            (fn [{:keys [success status body] :as response}]
-              (if (= status 409)
-                ;; Board name exists
-                (board-name-exists-error fixed-edit-key)
-                (create-update-entry-cb fixed-edited-data fixed-edit-key response)))))
-        (api/update-entry fixed-edited-data fixed-edit-key create-update-entry-cb))
-      (if (and (= (:board-slug fixed-edited-data) utils/default-section-slug)
-               section-editing)
-        (let [fixed-entry-data (dissoc fixed-edited-data :board-slug :board-name :invite-note)
-              final-board-data (assoc section-editing :entries [fixed-entry-data])]
-          (api/create-board final-board-data (:invite-note fixed-edited-data)
-            (fn [{:keys [success status body] :as response}]
-              (if (= status 409)
-                ;; Board name exists
-                (board-name-exists-error fixed-edit-key)
-                (create-update-entry-cb fixed-edited-data fixed-edit-key response)))))
-        (let [org-slug (router/current-org-slug)
-              entry-board-data (dis/board-data @dis/app-state org-slug (:board-slug fixed-edited-data))
-              entry-create-link (utils/link-for (:links entry-board-data) "create")]
-          (api/create-entry fixed-edited-data fixed-edit-key entry-create-link create-update-entry-cb))))
-    (dis/dispatch! [:entry-save fixed-edit-key])))
+(defn entry-save
+  ([edited-data section-editing]
```

Comment:
This function needs an `edit-key` as parameter so we know what key to use.

For example the component entry-edit uses `:entry-editing` to save the editing data but the new cmail uses `:cmail-data`. We use different keys because this app-state key is also used to decide which component we need to show.
In `org-dashboard` we show `entry-edit` if `:entry-editing` is not nil and `cmail` if `:cmail-data` is not nil.